lib/mapUtils.js

```diff
@@ -292,3 +298,131 @@ export function openDirections(latitude, longitude, storeName) {
+  const distance = convertDistance(distanceMeters, 'mi').toFixed(2);
+  return distance;
+}
+
+export async function orderStoresByDistance(region, inputStores) {
+  const sortedStores = [];
+  // Display distance in the StoreList
+  inputStores.forEach((store) => {
+    const currStore = store;
+    currStore.distance = findStoreDistance(region, store);
+    sortedStores.push(currStore);
+  });
+  // sorts in place
+  sortedStores.sort(function compare(a, b) {
+    return a.distance - b.distance;
+  });
+  return sortedStores;
+}
+
+export function findDefaultStore(stores) {
+  const defaultStore = stores.find((store) => {
+    return store.id === RecordIds.defaultStoreId;
+  });
+
+  const defaultRegion = {
+    latitude: defaultStore.latitude,
+    longitude: defaultStore.longitude,
+    ...deltas,
+  };
+  return { defaultStore, defaultRegion };
+}
+
+export function useCurrentLocation(setInitialLocation, setLocationPermissions) {
+  useEffect(() => {
+    const getCurrentLocation = async () => {
+      try {
+        const { status } = await Permissions.askAsync(Permissions.LOCATION);
+        if (status !== 'granted') {
+          Analytics.setUserProperty('location_permissions', 'denied');
+          setLocationPermissions('denied');
+        } else {
+          const location = await Location.getCurrentPositionAsync({
+            timeout: 3000,
+          });
+          const currentRegion = {
+            latitude: location.coords.latitude - deltas.latitudeDelta / 3.5,
+            longitude: location.coords.longitude,
+            ...deltas,
+          };
+          setInitialLocation(currentRegion);
+          setLocationPermissions('granted');
+          Analytics.setUserProperty('location_permissions', 'granted');
+        }
+      } catch (err) {
+        Analytics.setUserProperty('location_permissions', 'error');
+        logErrorToSentry({
+          screen: 'MapScreen',
+          function: 'useCurrentLocation',
+          error: err,
+        });
+        setLocationPermissions('error');
+      }
+    };
+    getCurrentLocation();
+  }, [setInitialLocation, setLocationPermissions]);
+}
+
+/**
+ * Sets initial stores in state.
+ */
+export function useInitialStores(setStores) {
+  useEffect(() => {
+    const populateInitialStores = async () => {
+      try {
+        // Sets list of stores in state, populates initial products
+        const storeData = await getStoreData();
+        setStores(storeData);
+      } catch (err) {
+        console.error('[MapScreen] (useInitialStores) Airtable:', err);
+        logErrorToSentry({
+          screen: 'MapScreen',
+          function: 'useInitialStores',
+          error: err,
+        });
+      }
+    };
+    populateInitialStores();
+  }, [setStores]);
+}
+/**
+ * If location permissions are granted and location is found, order stores by distance.
+ */
+export function useSortedStores(
+  locationPermissions,
+  stores,
+  setStores,
+  isSorted,
+  setSorted,
+  initialLocation
```

Comment:
This is one of the situations where there are SO MANY dependencies! This feels super excessive, but I _think_ I need these to make sure location/stores are fully loaded before attempting to sort.

`orderLoadedStores` only runs if:
- location permissions are granted
- stores are loaded
- stores haven't already been sorted
- user's location has been determined `initialLocation` is not `null`

Reply:
Also, I used `isSorted` to avoid this repeating forever, but it seems weird to keep since it's only purpose is to say that this function has successfully completed. Are there better ways to handle that?

Reply:
I would say that this could be a non-hook function that gets passed as a `.sort` callback once the stores have been loaded — this sorting seems like more of a transformation that can be done on-the-fly instead of updated in a stateful value.

That way you could use it like:

```javascript
const { loading, stores } = useStores(); // Or however you load in stores
if (loading) {
  // Return loading state
}

return stores.sort(orderByLocation).map((store) => <Store {...} />);
```

If the sorting function was costly or actually async (in my below comment, I called out that it might not have to be designated as async) — you could consider memoizing (with `useMemo`) to only run this transformation if the `stores` value changes.